### PR TITLE
CS lime-189, Inefficient Struct Packing

### PR DESCRIPTION
**Pull request by KalpitaMandal**
# Description

The struct `ExtensionVariables` has three variables of type `uint256`. At least the time and interval variables could be packed together to save a slot as it is highly unlikely, that these variables need a full word.

# Integrations Checklist

- [ ]  Have any function signatures changed? If yes, outline below.
- [ ]  Have any features changed or been added? If yes, outline below.
- [ ]  Have any events changed or been added? If yes, outline below.
- [ ]  Has all documentation been updated?

# Changelog

- Efficient struct packing in `Extension.sol`.

# Event Signature Changes

# Function Signature Changes

# Features

# Events

### CODE REVIEW

File: contracts/Pool/Extension.sol [hunk 1]
```diff
@@ -18,10 +18,10 @@ contract Extension is Initializable, IExtension {
 
     struct ExtensionVariables {
         bool hasExtensionPassed;
+        uint64 extensionVoteEndTime;
+        mapping(address => uint64) lastVotedExtension;
         uint256 totalExtensionSupport;
-        uint256 extensionVoteEndTime;
         uint256 repaymentInterval;
```
Comment (prateekreddy) [inline contracts/Pool/Extension.sol:24]: May be we can combine repaymentInterval as well with the first 2 vars, because bool takes 8 bits only.

**Review by prateekreddy — CHANGES_REQUESTED** [review]

File: contracts/Pool/Extension.sol [hunk 2]
```diff
@@ -18,10 +18,10 @@ contract Extension is Initializable, IExtension {
 
     struct ExtensionVariables {
         bool hasExtensionPassed;
+        uint64 extensionVoteEndTime;
+        mapping(address => uint64) lastVotedExtension;
```
Comment (prateekreddy) [inline contracts/Pool/Extension.sol:22]: mapping can be uint256 only, we will end up spending more gas if we use uint64

Comment (KalpitaMandal) [reply contracts/Pool/Extension.sol:22]: reverted to uint256

File: contracts/Pool/Extension.sol (hunk 1 above)
Comment (KalpitaMandal) [reply contracts/Pool/Extension.sol]: Packing `repaymentInterval` further using uint128

**Review by prateekreddy — APPROVED** [review]

**Pull request merged**